include/dxc/Support/WinAdapter.h

```diff
@@ -1039,8 +1040,62 @@ class CHandle {
   HANDLE m_h;
 };
 
+
+/////////////////////////////////////////////////////////////////////////////
+// CComBSTR
+
+class CComBSTR
+{
+public:
+    BSTR m_str;
+    bool m_needFree;
+    CComBSTR() : m_str(nullptr), m_needFree(false) {};
+    CComBSTR(_In_ int nSize, LPCWSTR sz);
+    ~CComBSTR() throw() {
+    if (m_needFree)
```

Comment:
Isn't this tied to whether `m_str` is non-null?  I don't think the independent variable is needed, since all BSTR strings must be freed with SysFreeString somehow.

Reply:
Removed m_needFree.